test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1

```diff
@@ -1,49 +1,140 @@
+            $result = Wait-Job -Timeout 2 $j
+            $result | Should Be $null
+        }
+        It "Stop-Job will stop a job" {
+            Stop-Job -Id $j.Id
+            $j.State | Should Be "Stopped"
+        }
+        It "Remove-Job will not remove a running job" {
+            $id = $j.Id
+            Remove-Job $j -ErrorAction SilentlyContinue
+            $job = Get-Job -Id $id
+            $job | Should Be $j
+        }
+        It "Remove-Job -Force will remove a running job" {
+            $id = $j.Id
+            Remove-Job $j -Force
+            $job = Get-Job -Id $id -ErrorAction SilentlyContinue
+            $job | Should Be $null
+        }
+    }
+    Context "Retrieving partial output from jobs" {
+        BeforeAll {
+            function GetResults($job, $n, $keep)
+            {
+                $results = @()
+
+                # $count allows us to bail out after 5 minutes, avoiding an endless loop
+                for ($count = 0; $results.Count -lt $n; $count++)
+                {
+                    if ($count -eq 1000)
+                    {
+                        # It's been 5 minutes and we still have collected enough results!
+                        throw "Receive-Job behaves suspiciously: Cannot receive $n results in 5 minutes."
+                    }
+
+                    # sleep for 300 ms to allow data to be produced
+                    Start-Sleep -Milliseconds 300
+
+                    if ($keep)
+                    {
+                        $results = Receive-Job -Keep $job
+                    }
+                    else
+                    {
+                        $results += Receive-Job $job
+                    }
+                }
+
+                return $results
+            }
+
+            function CheckContent($array)
+            {
+                for ($i=1; $i -lt $array.Count; $i++)
+                {
+                    if ($array[$i] -ne ($array[$i-1] + 1))
+                    {
+                        return $false
+                    }
+                }
+
+                return $true
+            }
+
+        }
+        BeforeEach {
+            $j = Start-Job -ScriptBlock { $count = 1; while ($true) { Write-Output ($count++); Start-Sleep -Milliseconds 30 } }
+        }
+        AfterEach {
+            Get-Job | Remove-Job -Force
+        }
+
+        It "Receive-Job will retrieve partial output" {
+            $result1 = GetResults $j 5 $false
+            $result2 = GetResults $j 5 $false
+            CheckContent ($result1 + $result2)
         }
-        It "Stop-Job will stop a job" -pending {
-            Stop-Job -id $j.id
-            $j.Status |Should be Stopped
+        It "Receive-Job will retrieve partial output, including -Keep results" {
+            $result1 = GetResults $j 5 $true
+            $result2 = GetResults $j ($result1.Count + 5) $false
+            Compare-Object -PassThru $result1 $result2[0..($result1.Count-1)] | Should Be $null
```

Comment:
With `-SyncWindow 0`, `Compare-Object` compares 2 lists in order. Example:
```
PS> $a = 1,2,3,4
PS> $b = 2,3,4,1
PS> $c = 1,2,3,4
PS> Compare-Object -ReferenceObject $a -DifferenceObject $b
PS> Compare-Object -ReferenceObject $a -DifferenceObject $b -SyncWindow 0

InputObject SideIndicator
----------- -------------
          2 =>
          1 <=
          3 =>
          2 <=
          4 =>
          3 <=
          1 =>
          4 <=

PS> Compare-Object -ReferenceObject $a -DifferenceObject $c -SyncWindow 0
PS>
```
You can add this parameter if you want to compare them in order.